src/Chainweb/HostAddress.hs

```diff
@@ -232,7 +219,7 @@ ipV6Parser = p0
 
 portParser :: Parser Port
 portParser = Port
-    <$> (decimal >>= \(d :: Integer) -> int d <$ guard (d < 2^(16 :: Int)))
+    <$> (decimal >>= \(d :: Integer) -> int d <$ guard (d <= 2^(16 :: Int)-1))
```

Comment:
isn't this an equivalent statement?

Reply:
it is for `d :: Integer`. I got bitten by copy and pasting it to a context with `d :: Word16` in which case it isn't equivalent. So, I thought it was better to use the more robust code.

Reply:
ahh interesting alright!